frontend/src/components/auth/usePageTitle.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useRef } from "react";
+
+const usePageTitle = (
+  title?: string,
+  skip = false,
+  setOnlyIfUnchanged = false,
+) => {
+  const prevTitle = useRef("");
+  prevTitle.current = document.title;
+  useEffect(() => {
+    if (skip) return;
+    if (!title) return;
+    if (setOnlyIfUnchanged && prevTitle.current !== document.title) return;
+    document.title = `${title} | Jump Math`;
```

Comment:
do we need to update `prevTitle.current` when title changes? not exactly sure what this logic is doing